knowledge-base/adr/0020-generator-options.md

```diff
@@ -0,0 +1,57 @@
+| `--generatePackageJson`                                  | `--packageJson`                  | Writes a default package.json, default true.                                                      | Writes a default package.json, default true (needs core dependency). Writes a custom package.json if passed (or use `copyFiles`). |    true    |
+| `--serviceMapping`                                       | `--serviceMapping`               | Considers a service mapping file. Shows a warning if none is given (differs from OData behavior). | Considers a service mapping file. Generates it by default. Skips if set to false.                                                 |    true    |
+| `--tsConfig`                                             | `--tsConfig`                     | tsconfig.json file to overwrite the default "tsconfig.json".                                      | Writes a default tsconfig.json, default false. Writes a custom tsconfig.json if passed (or use `copyFiles`).                      |   maybe?   |
+| `--versionInPackageJson`                                 | `--packageVersion`               | Version in package.json, default is generator version.                                            | Version in package.json, default is `1.0.0`.                                                                                      |   maybe?   |
+| **Options in OData, but not (yet in OpenAPI) generator** |
+| `--forceOverwrite`                                       | `--forceOverwrite`, `-f`         | Overwrite files even it they exist, default false.                                                | Overwrite files even it they exist, default false.                                                                                |   maybe?   |
+| `--generateTypedocJson`                                  | `--typedocJson`                  | Writes a default typedoc.json, default true.                                                      | Remove, use `copyFiles` instead.                                                                                                  |   maybe?   |
+| **Currently hidden options**                             |
+| `--additionalFiles`                                      | `--copyFiles`                    | Copy additional files, identified by glob. Hidden.                                                | Copy additional files, identified by glob. Not hidden.                                                                            |   maybe?   |
+| `--writeReadme`                                          | `--readme`                       | Writes a default README.md, default false.                                                        | Writes a default README.md, default false. Writes a custom readme if passed (or use `copyFiles`).                                 |   maybe?   |
```

Comment:
For us, the links of the README should be tested no matter we have default README (current) or use `copyFiles` with a template.

Reply:
Yes, this can be done independently, correct?